gcs/src/gcs_group.cpp

```diff
@@ -747,6 +747,19 @@ gcs_group_handle_join_msg  (gcs_group_t* group, const gcs_recv_msg_t* msg)
 
             if (from_donor && peer_idx == group->my_idx &&
                 GCS_NODE_STATE_JOINER == group->nodes[peer_idx].status) {
+
+                // If there is an ENODATA error code, then it is indication
+                // that state on the joiner node was moved forward too much
+                // and we need to initiate a full SST instead of IST.
+                // We should not treat this condition as a fatal error
+                // (decision on this matter should take higher levels,
+                // for example, Replicator or the server):
+
+                if (seqno == -ENODATA)
```

Comment:
PXC#426 we need to fix only heuristic and if SST still fails then we need to let the server shutdown with proper error message. This return will not let it happen. Instead it will let it continue and as per the comment initiate a full SST.